examples/with-now-env/next.config.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * After the next require you can use process.env to get your secrets
+ */
+require('now-env')
+
+console.log({
+  SECRET: process.env.SECRET,
+  ANOTHER_SECRET: process.env.ANOTHER_SECRET,
+  SECRET_FAIL: process.env.SECRET_FAIL
+})
+
+/**
+ * If some of the envs are public, like a google maps key, but you still
+ * want to keep them secret from the repo, the following code will allow you
+ * to share some variables with the client, configured at compile time.
+ */
+module.exports = {
+  webpack: config => {
+    config.plugins.forEach(plugin => {
+      if (plugin.constructor.name === 'DefinePlugin') {
+        plugin.definitions['process.env.SECRET'] = JSON.stringify(process.env.SECRET)
```

Comment:
Why not just do another defineplugin, is much more readable and future proof.

Reply:
Instead of `DefinePlugin` you could have use `EnvironmentPlugin` and just do:

```js
new webpack.EnvironmentPlugin(['SECRET'])
```

That will automatically define `process.env.SECRET` 😉

Reply:
@sergiodxa I like explicit mapping though. Especially since this ends up in client bundles.